resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/SparkRackResolver.scala

```diff
@@ -70,18 +70,20 @@ private[spark] class SparkRackResolver(conf: Configuration) extends Logging {
     val nodes = new ArrayBuffer[Node]
     // dnsToSwitchMapping is thread-safe
     val rNameList = dnsToSwitchMapping.resolve(hostNames.toList.asJava).asScala
-    if (rNameList == null || rNameList.isEmpty) {
+    if (rNameList == null) {
       hostNames.foreach(nodes += new NodeBase(_, NetworkTopology.DEFAULT_RACK))
       logInfo(s"Got an error when resolving hostNames. " +
         s"Falling back to ${NetworkTopology.DEFAULT_RACK} for all")
     } else {
-      for ((hostName, rName) <- hostNames.zip(rNameList)) {
-        if (Strings.isNullOrEmpty(rName)) {
-          nodes += new NodeBase(hostName, NetworkTopology.DEFAULT_RACK)
-          logDebug(s"Could not resolve $hostName. " +
-            s"Falling back to ${NetworkTopology.DEFAULT_RACK}")
-        } else {
-          nodes += new NodeBase(hostName, rName)
+      if (rNameList.nonEmpty) {
+        for ((hostName, rName) <- hostNames.zip(rNameList)) {
```

Comment:
if `hostNames` is nonEmpty, but `rNameList` is empty, this will just silently skip everything, right?  that shouldn't happen, but as we're calling code outside of spark a little defensiveness would be good here.  I think the calling code requires the returned list to be the same length as `hostNames`.

I am also biased against early returns, but here I think it really is the simplest alternative.

(sorry for so much quibbling about such a small thing ...)

Reply:
Ooooh gosh! This PR is not something which I'm telling tales to my children :)
I was sooo focused on the if condition which wrapped the log that I've mixed up `rNameList` and `hostNames`. This case I agree on having an early return.